### PR TITLE
Added ISO management

### PR DESCRIPTION
**Pull request by joe-singleton-itdev**
As per issue #51

### CODE REVIEW

**Comment by itdevreviewboard** [comment]
Can one of the MCVirt admins verify this patch?

File: source/usr/lib/mcvirt/iso.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                chunk = iso.read(chunk_size)
+                if not chunk:
+                    break
+                file.write(chunk)
+                
+        IsoManagement.setIsoPermissions(output_path)
+        
+        return 'ISO downloaded as %s' % name
+          
+class InvalidISOPathException(MCVirtException):
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/iso.py]: Generally, we put exception classes at the top of the module

File: source/usr/lib/mcvirt/parser.py [hunk 2]
```diff
@@ -719,3 +731,22 @@ def parse_arguments(self, script_args=None, mcvirt_instance=None):
 
         elif (action == 'list'):
             mcvirt_instance.listVms()
+        
+        elif (action == 'iso'):
+            output = ''
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/parser.py]: I would say, set this initially to None

File: source/usr/lib/mcvirt/parser.py [hunk 3]
```diff
@@ -81,6 +82,17 @@ def __init__(self, print_status=True):
                                                       parents=[self.parent_parser])
         self.stop_parser.add_argument('vm_name', metavar='VM Name', type=str, help='Name of VM')
 
+
+        # Add arguments for ISO functions
+        self.iso_parser = self.subparsers.add_parser('iso', help='ISO managment',
+                                                     parents=[self.parent_parser])
+        self.iso_parser.add_argument('--list', dest='list', action='store_true', help='List available ISOs')
+        self.iso_parser.add_argument('--add-from-path', dest='add_path', help='Copy an ISO to ISO directory',
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/parser.py]: Maybe say 'Import an ISO' or something? Or 'Add an ISO to MCVirt'

File: source/usr/lib/mcvirt/iso.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        # Get files in ISO directory
+        listing = os.listdir(mcvirt_instance.ISO_STORAGE_DIR)
+        for file in listing:
+            # If is a file and ends in '.iso' ...
+            if (os.path.isfile(mcvirt_instance.ISO_STORAGE_DIR + '/' + file) and IsoManagement.checkName(file)):
+                if len(list) > 0:
+                    list += '\n'
+                list += file
+                
+        if (len(list) == 0):
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/iso.py]: This could be simplified to:

    if (not len(list)):

**Comment by MatthewJohn** [comment]
Ready for Jenkins build

File: source/usr/lib/mcvirt/iso.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+# You should have received a copy of the GNU General Public License
+# along with MCVirt.  If not, see <http://www.gnu.org/licenses/>
+
+import os
+import stat
+
+from mcvirt import MCVirtException
+from system import System
+
+class IsoManagement:
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/iso.py]: I would just call this class 'Iso' :smiley:

File: source/usr/lib/mcvirt/iso.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        """Check that name ends in .iso"""
+        return name[-4:].lower() == '.iso'
+        
+    @staticmethod
+    def getFilename(path):
+        """Return filename part of path"""
+        return path.split('/')[-1]
+        
+    @staticmethod
+    def setIsoPermissions(path):
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/iso.py]: Given that this function is within the 'Iso' class, I think naming this to simply 'setPermissions' would be good... which probably applies to getIsoList -> getList, addIso -> add, deleteIso -> delete etc :)

**Comment by MatthewJohn** [comment]
This is ok to test

**Comment by MatthewJohn** [comment]
This has been superseded by #85

**Pull request closed without merging**